packages/api/endpoints/reconciliation-reports.js

```diff
@@ -117,27 +134,42 @@ async function getReport(req, res) {
  */
 async function deleteReport(req, res) {
   const name = req.params.name;
-  const reconciliationReportModel = new models.ReconciliationReport();
-  const record = await reconciliationReportModel.get({ name });
+  try {
+    const reconciliationReportPgModel = new ReconciliationReportPgModel();
+    const knex = await getKnexClient();
+    const record = await reconciliationReportPgModel.get(knex, { name });
+    if (!record.location) {
+      return res.boom.badRequest('The reconciliation report record does not contain a location!');
+    }
+    const { Bucket, Key } = parseS3Uri(record.location);
 
-  const { Bucket, Key } = parseS3Uri(record.location);
-  if (await fileExists(Bucket, Key)) {
-    await deleteS3Object(Bucket, Key);
-  }
-  await reconciliationReportModel.delete({ name });
-
-  if (inTestMode()) {
-    const esClient = await getEsClient(process.env.ES_HOST);
-    await indexer.deleteRecord({
-      esClient,
-      id: name,
-      type: 'reconciliationReport',
-      index: process.env.ES_INDEX,
-      ignore: [404],
-    });
+    await createRejectableTransaction(knex, async (trx) => {
+      if (await fileExists(Bucket, Key)) {
+        await deleteS3Object(Bucket, Key);
+      }
+      await reconciliationReportPgModel.delete(knex, { name });
+    })
+
+    if (inTestMode()) {
+      const esClient = await getEsClient(process.env.ES_HOST);
+      await indexer.deleteRecord({
+        esClient,
+        id: name,
+        type: 'reconciliationReport',
+        index: process.env.ES_INDEX,
+        ignore: [404],
+      });
+    }
+
+    return res.send({ message: 'Report deleted' });
+  } catch (error) {
```

Comment:
The try/catch covers lots of statements, but the `RecordDoesNotExist` only comes from
`const record = await reconciliationReportPgModel.get(knex, { name });`
suggest to add try/catch around that statement.

Reply:
I've adjusted the try/catch to be closer to what it was before, removed the unreachable statement, and changed the last successful statement

Reply:
So what happens if there is an error after that, as in from the `parseS3Uri` or the `delete` operations? You want to not try and catch an error for those?

Reply:
You only handle RecordDoesNotExist error which comes from `await reconciliationReportPgModel.get(knex, { name });`